File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        l,r = 0, len(height) -1
        lm, rm = height[l], height[r]
        total = 0
        while l <= r:
            if lm < rm:
                lm = max(lm, height[l])
                total += lm - height[l]
                l += 1
            else:
                rm = max(rm, height[r])
                total += rm - height[r]
                r -= 1
        return total



            
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for (int n : nums) {
            numSet.add(n);
        }

        int longest = 0;
        for (Integer n : numSet) {
            if (!numSet.contains(n-1)) {
                int length = 1;
                for (;numSet.contains(n + length); length++);
                longest = Math.max(longest, length);
            }
        }
        return longest;
    }
}
